### PR TITLE
Hshami/py up

### PR DESCRIPTION
**Pull request by mhshami01**
remove support for py27

### CODE REVIEW

File: iotedgehubdev/composeproject.py [hunk 1]
```diff
@@ -4,11 +4,11 @@
 
 import json
 import os
+import io
```
Comment (smcd253) [inline iotedgehubdev/composeproject.py:7]: StringIO() is the only function used here, so you can change to "from io import StringIO" to stay consistent with iotedgedev

File: requirements.txt [hunk 2]
```diff
@@ -3,18 +3,16 @@ backports.unittest-mock
...
+pyOpenSSL>=20.0.1
 python-dotenv
 requests>=2.25.1
-six
 applicationinsights
 prompt_toolkit
 rope
 tox
-pyyaml==5.4
+pyyaml==5.3.1
```
Comment (smcd253) [inline requirements.txt:13]: I think there's a security vulnerability in this version specifically. Can you pick another version or are we locked into 5.3.1?

File: setup.py [hunk 3]
```diff
@@ -58,7 +56,7 @@
             'iotedgehubdev = iotedgehubdev.cli:main',
         ],
     },
-    python_requires='>=2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*, !=3.4.*, <3.9',
+    python_requires='>=3.5, <3.9',
```
Comment (smcd253) [inline setup.py:59]: Are we testing with 3.5, 3.6, 3.7, and 3.8?

File: requirements.txt (hunk 2 above)
Comment (cindydeng1998) [inline requirements.txt:13]: why are we going down to a lower version of pyyaml here?

File: setup.py [hunk 4]
```diff
@@ -76,8 +74,6 @@
         'Operating System :: Unix',
         'Operating System :: Microsoft :: Windows',
         'Programming Language :: Python',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
```
Comment (cindydeng1998) [inline setup.py:79]: Do we need to add 3.7 and 3.8 here

File: tests/test_compose_resources/docker-compose.yml [hunk 5]
```diff
@@ -1,77 +1,77 @@
...
+    - routes__r1=FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO
+      BrokeredEndpoint("/modules/SampleModule/inputs/input1")
+    - routes__r2=FROM /messages/modules/SampleModule/outputs/output1 INTO $$upstream
+    - IotHubConnectionString=HostName=HostName;DeviceId=DeviceId;ModuleId=$$edgeHub;SharedAccessKey=SharedAccessKey
+    - EdgeModuleHubServerCAChainCertificateFile=/mnt/edgehub/edge-chain-ca.cert.pem
+    - EdgeModuleHubServerCertificateFile=/mnt/edgehub/edge-hub-server.cert.pfx
+    - configSource=local
+    - SSL_CERTIFICATE_PATH=/mnt/edgehub/
+    - SSL_CERTIFICATE_NAME=edge-hub-server.cert.pfx
+    image: mcr.microsoft.com/azureiotedge-hub:1.1
```
Comment (cindydeng1998) [inline tests/test_compose_resources/docker-compose.yml:16]: on edgedev the edgehub version tag here is swappable for both 1.0 and 1.1, do we want to default to 1.1 in unit tests?

**Review by marianan — APPROVED** [review]
:shipit:

File: setup.py (hunk 4 above)
Comment (mhshami01) [reply setup.py:79]: 3.7 and 3.8 are there. I don't know why they don't show in the diff

File: requirements.txt (hunk 2 above)
Comment (mhshami01) [reply requirements.txt:13]: that's the highest we can go to support py35

**Pull request merged**